### PR TITLE
OpInfo porting for msort operator

### PR DESCRIPTION
**Pull request by iramazanli**
Fixes #{issue number}

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -3841,6 +3848,11 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                SkipInfo('TestOperatorSignatures', 'test_get_torch_func_signature_exhaustive'),
            ),
            sample_inputs_func=sample_inputs_unfold),
+    OpInfo('msort',
+           dtypes=all_types_and(torch.bool, torch.float16),
+           supports_out=False,
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3853]: msort does support out though

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -3841,6 +3848,11 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                SkipInfo('TestOperatorSignatures', 'test_get_torch_func_signature_exhaustive'),
            ),
            sample_inputs_func=sample_inputs_unfold),
+    OpInfo('msort',
+           dtypes=all_types_and(torch.bool, torch.float16),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3852]: test failures suggest msort does not support bool

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -3875,6 +3881,10 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
                SkipInfo('TestOperatorSignatures', 'test_get_torch_func_signature_exhaustive'),
            ),
            sample_inputs_func=sample_inputs_unfold),
+    OpInfo('msort',
+           dtypes=all_types_and(torch.float16),
+           check_batched_gradgrad=False,
+           sample_inputs_func=sample_inputs_msort),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3887]: Add a skip for test_out and add a comment above the skip with the error message.

Looks like msort does support bool, but only on a certain inputs? If that's the case then remove it from supported dtypes (like the current OpInfo) and skip the supported dtypes test for bool.

Are you able to run the msort opinfo tests locally?

Comment (iramazanli) [reply torch/testing/_internal/common_methods_invocations.py:3981]: Yes I can actually run tests locally. and I think they can pass all of the tests now.

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -3961,6 +3967,18 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+    OpInfo('msort',
+           dtypes=all_types_and(torch.float16),
+           check_batched_gradgrad=False,
+           skips=(
+               #  msort does not correctly warn when resizing out= inputs.
+               SkipInfo('TestCommon', 'test_out',
+                        dtypes=all_types_and_complex_and(torch.bool, torch.float16, torch.bfloat16)),
+               #  msort does not raise expected Runtime Error.
+               SkipInfo('TestOpInfo', 'test_unsupported_dtypes',
+                        dtypes=all_types_and_complex_and(torch.bool)),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:3979]: Does this skip need to need for all dtypes or only for torch.bool?

Comment (iramazanli) [reply torch/testing/_internal/common_methods_invocations.py]: You're right as always :) torch.bool is actually enough

**Review by mruberry — APPROVED** [review]
Cool! This looks good. Let's get it landed.

**Pull request closed without merging**